packages/@ama-mfe/ng-utils/src/managers/interfaces.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+  Message,
+  RoutedMessage,
+} from '@amadeus-it-group/microfrontends';
+import type {
+  ErrorContent,
+} from '../messages/index';
+
+/**
+ * Use the message payload to execute an action based on message type
+ * @param message message to consume
+ */
+export type MessageCallback<T extends Message> = (message: RoutedMessage<T>) => void | Promise<void>;
+
+/**
+ * Description of the accepted messages versions and their callbacks for a given message type
+ */
+export interface MessageVersions<T extends Message> {
+  /** The message callback coresponding to the version */
+  [version: string]: MessageCallback<T>;
+}
+
+/** Description for the base message consumer */
+export interface BasicMessageConsumer {
+  /** Contains the message type */
+  type: string;
+  /** A map of messages versions and their callbacks */
+  supportedVersions: Record<string, MessageCallback<any>>;
+}
```

Comment:
Shoudn't we have something like this ?
```suggestion
export interface BasicMessageConsumer<T extends Message = Message> {
  /** Contains the message type */
  type: T['type'];
  /** A map of messages versions and their callbacks */
  supportedVersions: Record<string, MessageCallback<T>>;
}
```